drivers/storage/partmgr/partition.c

```diff
@@ -25,10 +25,12 @@ PartitionCreateDevice(
 
     WCHAR nameBuf[64];
     UNICODE_STRING deviceName;
+    UINT32 volumeNum;
 
     // create the device object
 
-    swprintf(nameBuf, L"\\Device\\HarddiskVolume%u", HarddiskVolumeNextId++);
+    volumeNum = HarddiskVolumeNextId++;
+    swprintf(nameBuf, L"\\Device\\HarddiskVolume%u", volumeNum);
```

Comment:
Shouldn't we use `%lu` here by the way?

Reply:
So the compiler does not yell at you or should this be POSIX compatible? unsigned long and unsigned int are obviously the same on Windows.

Reply:
That would be preferable over %u.